### PR TITLE
initial release

### PR DESCRIPTION
**Pull request by robo-cap**

### CODE REVIEW

File: locals_vulnerability_scan_policies.tf [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+locals {
+  vulnerability_scan_policy_params = {
+    container_image_scanning_policy = {
+      compartment_name  = "tenancy"
+      name              = "container_image_scanning_policy"
+      description       = "Container Image Registry scanning policies"
+      statements        = ["Allow service vulnerability-scanning-service to read repos in tenancy",
+                          "Allow service vulnerability-scanning-service to read compartments in tenancy"
+      ]
+      defined_tags      = merge(var.default_defined_tags, { "${oci_identity_tag_namespace.ArchitectureCenterTagNamespace.name}.${oci_identity_tag.ArchitectureCenterTag.name}" = var.release })
```
Comment (mp3monster) [inline locals_vulnerability_scan_policies.tf:13]: there is a library in the repo for generating the defined tags for you

File: locals_rules.tf [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      compartment_name = "monitored_compartment"
+      is_enabled       = true
+      condition        = "{\"eventType\":[\"com.oraclecloud.computeapi.launchinstance.end\"]}"
+      actions          = [{
+        is_enabled       = true
+
+        is_invoke_function   = true
+        invoke_function_name = "dns_record_add"
+      }]
+      defined_tags     = merge(var.default_defined_tags, { "${oci_identity_tag_namespace.ArchitectureCenterTagNamespace.name}.${oci_identity_tag.ArchitectureCenterTag.name}" = var.release })
```
Comment (mp3monster) [inline locals_rules.tf:18]: there is a library in the repo for generating the defined tags for you

File: modules/events/main.tf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  
+  actions {
+    dynamic "actions" {
+      for_each   = each.value.actions
+      iterator   = action
+      content {
+          action_type = ( action.value.is_invoke_function == true ?
+                            "FAAS" :
+                            action.value.is_post_to_stream == true ?
+                              "OSS" :
```
Comment (mp3monster) [inline modules/events/main.tf:23]: minor issue - might want to tidy so line up

File: modules/functions/main.tf [hunk 4]
```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+locals {
+  ocir_repository_domain = join("", [lower(lookup(data.oci_identity_regions.oci_regions.regions[0], "key")), ".ocir.io"])
+  ocir_namespace         = lookup(data.oci_objectstorage_namespace.os_namespace, "namespace")
+  function_ocir_uri      = { for k, v in var.function_params:
```
Comment (mp3monster) [inline modules/functions/main.tf:7]: tnis is pretty dense code to read - worthy of a simple comment to explain what is happening

File: modules/functions/main.tf [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+  for_each                    = { for k, v in var.application_params: 
+                                  k => v 
+                                  if v.enable_oci_logging == true
+  }
+  
+  compartment_id = var.compartment_ids[each.value.compartment_name]
+}
+
+
+resource "oci_logging_log_group" "this" {
```
Comment (mp3monster) [inline modules/functions/main.tf:153]: this looks very reusable - suggest consider making it a module

File: schema.yaml [hunk 6]
```diff
@@ -0,0 +1,127 @@
+title: DNS Automation using functions
```
Comment (mp3monster) [inline schema.yaml:1]: copyright as a comment

File: README.md [hunk 7]
```diff
@@ -1,32 +1,81 @@
...
 
-## Notes/Issues
-MISSING
 
+## Manual deployment
+
+ **Prerequisites:** `bash`, `terraform`, `fn`, `docker`
+
+1. Create a file named `terraform.auto.tfvars` in the root directory using below list of variables and update associated values based on your use-case:
+```
```
Comment (mp3monster) [inline README.md:37]: consider adding link to OCI docs describing where to get fimgerprint etc

File: README.md [hunk 8]
```diff
@@ -1,32 +1,81 @@
...
 ## Introduction
-MISSING
+
+The DNS Private Zone feature within OCI's DNS service allows the users to expand the capabilities of the built-in VCN DNS resolver, enabling them to use custom domain names for the resources within the VCNs.
+
+This sample deployment uses OCI Events to monitor a parent compartment for compute resource creation and termination. When such an event occurs, OCI Functions are triggered to maintain the records in a user-configured DNS Zone. 
+
+In case of duplicate records, the function will attempt to generate a new hostname using the latest five characters in instance OCID.
+
+Two free-form tags: `hostnames` and `dns_zone_record_hashes`, will be attached to the instance to confirm the DNS Zone record insertion at instance creation and DNS Zone record deletion at instance termination.
```
Comment (mp3monster) [inline README.md:13]: we started to adopt a pattern of also adding free form tags naming the module/ repo that was used as the source.  Helped back track if there was a need to help a customer. Plus if a TF template was updated the customer coult determine if they have deployed it

File: tags.tf [hunk 9]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+resource "random_id" "tag" {
```
Comment (mp3monster) [inline tags.tf:4]: there is a module that will sort out random values

File: schema.yaml (hunk 6 above)
Comment (robo-cap) [reply schema.yaml:1]: added

**Review by mp3monster — COMMENTED** [review]
provided review comments - nothing fundamentally wrong. But somethings would add to maintainability

**Review by mp3monster — APPROVED** [review]

**Pull request merged**